www/js/insert-slide.js

```diff
@@ -0,0 +1,188 @@
+  '<td>',
+  '<span class="lang-text">Announcement in Gurmukhi?</span>',
+  '</td>',
+  '<td>',
+  '<div class="switch">',
+  '<input id="modal-ann-lang" name="modal-ann-lang" type="checkbox" value="gurmukhi">',
+  '<label for="modal-ann-lang"></label>',
+  '</div>',
+  '</td>',
+  '</tr>',
+  '</table>',
+  '</div>',
+);
+const modalAnBox =
+  '<div class="box-container"><div class="modal-ann-box" contenteditable="true" data-placeholder="Add announcement text here ..."></div></div>';
+/**
+ * sets each dhan guru button an onclick that will send that guru's name to the slide
+ */
+function buttonOnClick() {
+  if (!isAnnouncementTab) {
+    for (let i = 1; i <= 11; i += 1) {
+      document.getElementById(`guru${i}`).onclick = () => {
+        global.controller.sendText(strings.slideStrings.dhanguruStrings[i - 1], true);
+        modal.close();
+      };
+    }
+  }
+}
+/**
+ * sets the onclick val for the slider
+ * appends class gurmukhi to the text box (changes font family to gurbaniakhar in css file)
+ * changes placeholder text
+ */
+function setLangSliderVal() {
+  const slider = document.querySelector('#modal-ann-lang');
+  slider.onclick = () => {
+    // is the slider checkd?
+    const isGurmukhi = slider.checked;
+    const placeholderText = isGurmukhi ? 'GoSxw ie`Qy ilKo ...' : 'Add announcement text here ..';
+
+    const $announcementBox = document.querySelector('.modal-ann-box');
+    $announcementBox.classList.toggle('gurmukhi', isGurmukhi);
+    $announcementBox.setAttribute('data-placeholder', placeholderText);
+  };
+}
+
+/**
+ * sanitize the html, allow only the permitted tage from allowedTags array above
+ */
+function boxInputFunctionality() {
+  const box = document.querySelector('.modal-ann-box');
+  box.oninput = () => {
+    const $announcementInput = document.querySelector('.announcement-text');
+    $announcementInput.innerHTML.replace(
+      /.*/g,
+      sanitizeHtml($announcementInput.innerHTML, { allowedTags }),
+    );
+  };
+}
+// sets the first tab, with header, section title, and button group
+const slidePage = slideHeader + slideText + buttons;
+// sets the second page with header, change language button, and content box
+const announcementPage = announcementHeader + langSlider + modalAnBox;
+// sets the default page to Dhan Guru slide page
+modal.setContent(slidePage);
+
+// sets the first button (OK) which on click sends the announcement to the screen
+modal.addFooterBtn('Ok', 'tingle-btn tingle-btn--pull-right tingle-btn--default', () => {
+  if (isAnnouncementTab) {
+    analytics.trackEvent('display', 'announcement-slide');
+    const isGurmukhi = document.querySelector('#modal-ann-lang').checked;
+    const announcementText = sanitizeHtml(document.querySelector('.modal-ann-box').innerHTML, {
+      allowedTags,
+    });
+    global.controller.sendText(announcementText, isGurmukhi);
+    document.querySelector('.modal-ann-box').innerHTML = '';
+    modal.close();
+  } else {
+    modal.close();
```

Comment:
can move this after the `if` and just write it once for both cases and not have an `else` .
```
if (isAnnouncementTab) {
   ....
} 
modal.close();
```